Makefile

```diff
@@ -1,231 +1,148 @@
-REPO            ?= vernemq
-PKG_REVISION    ?= $(shell git describe --tags)
-PKG_BUILD        = 1
 BASE_DIR         = $(shell pwd)
 ERLANG_BIN       = $(shell dirname $(shell which erl))
+GIT_VERSION      = $(shell git describe --tags)
+NOW              = $(shell date +%s)
+OS               = $(shell uname -s)
 OVERLAY_VARS    ?=
 REBAR ?= $(BASE_DIR)/rebar3
 
 $(if $(ERLANG_BIN),,$(warning "Warning: No Erlang found in your path, this will probably not work"))
 
-.PHONY: rel docs
 
 all: compile
 
 compile:
 	$(REBAR) $(PROFILE) compile
 
-deps:
-	$(REBAR) $(PROFILE) deps
-
-install_deps:
-	$(REBAR) $(PROFILE) install_deps
-
-clean: testclean
-	@rm -rf ebin
-
-distclean: clean relclean ballclean
-	@rm -rf _build
 
 rpi32: PROFILE = as rpi32
 rpi32: rel
 
 
-##
-## Test targets
-##
-TEST_LOG_FILE := eunit.log
-testclean:
-	@rm -f $(TEST_LOG_FILE)
-
-# Test each dependency individually in its own VM
-test: compile testclean
-	@$(foreach dep, \
-		$(wildcard _build/default/lib/*), \
-		(./rebar3 eunit --app=$(notdir $(dep)))  \
-		|| echo "Eunit: $(notdir $(dep)) FAILED" >> $(TEST_LOG_FILE);)
-	@if test -s $(TEST_LOG_FILE) ; then \
-		cat $(TEST_LOG_FILE) && \
-             exit `wc -l < $(TEST_LOG_FILE)`; \
-        fi
-
 ##
 ## Release targets
 ##
 rel:
 ifeq ($(OVERLAY_VARS),)
+	cat vars.config > vars.generated
 	$(REBAR) $(PROFILE) release
 else
-	cat vars.config > vars_pkg.config
-	cat $(OVERLAY_VARS) >> vars_pkg.config
-	$(REBAR) $(PROFILE) release --overlay_vars vars_pkg.config
-	cp _build/default/rel/vernemq/bin/start_clean.boot _build/default/rel/vernemq/releases/$(MAJOR_VERSION)/start_clean.boot
+	cat vars.config > vars.generated
+	cat $(OVERLAY_VARS) >> vars.generated
+	$(REBAR) $(PROFILE) release
 endif
 
-
-
-relclean:
-	rm -rf _build/default/rel
-
 ##
-## Doc targets
+## Support RPM and Debian based linux systems
 ##
-docs: compile
-	(cd docs && make clean && make html)
+ifeq ($(OS),Linux)
+ARCH          = $(shell uname -m)
+ISRPM         = $(shell cat /etc/redhat-release 2> /dev/null)
+ISDEB         = $(shell cat /etc/debian_version 2> /dev/null)
+ISSLES        = $(shell cat /etc/SuSE-release 2> /dev/null)
+ifneq ($(ISRPM),)
+PKGTARGET      = rpm
+else
+ifneq ($(ISDEB),)
+PKGTARGET      = deb
+else
+ifneq ($(ISSLES),)
+PKGTARGET      = rpm
+endif  # SLES
+endif  # deb
+endif  # rpm
+endif # linux
+
+package: $(PKGTARGET)
+
+pkg_rel: pkg_clean
+	$(MAKE) rel
+
+pkg_strip:
+	$(shell find _build/default/rel/vernemq -type f -executable | xargs strip --remove-section=.comment --remove-section=.note --strip-unneeded)
+
+deb: OVERLAY_VARS=vars/deb_vars.config
```

Comment:
make deb
rm -rf _build/default/rel
make rel
make[1]: Entering directory '/home/lhc/dev/octavolabs/vernemq'
cat vars.config > vars.generated
cat vars/deb_vars.config >> vars.generated
/home/lhc/dev/octavolabs/vernemq/rebar3  release
===> Verifying dependencies...
===> Compiling vmq_ql
===> Running neotoma...
===> Compiling vmq_plugin
===> Compiling vmq_commons
===> Compiling vmq_server
===> Compiling vmq_webhooks
===> Compiling vmq_swc
===> Compiling vmq_acl
===> Compiling vmq_plumtree
===> Compiling vmq_passwd
make[2]: Entering directory '/home/lhc/dev/octavolabs/vernemq/apps/vmq_passwd/c_src'
cc   vmq_passwd.c -lcrypto -o ../priv/vmq_passwd
make[2]: Leaving directory '/home/lhc/dev/octavolabs/vernemq/apps/vmq_passwd/c_src'
===> Compiling vmq_mqtt5_demo_plugin
===> Compiling vmq_bridge
===> Compiling vmq_diversity
===> Running cuttlefish schema generator
===> Starting relx build process ...
===> Resolving OTP Applications from directories:
          /home/lhc/dev/octavolabs/vernemq/_build/default/lib
          /home/lhc/dev/octavolabs/vernemq/apps
          /home/opt/erlang/erlang21.3.1/lib
===> Resolved vernemq-1.7.0+build.34.ref4174d018
===> Including Erts from /home/opt/erlang/erlang21.3.1
===> release successfully created!
make[1]: Leaving directory '/home/lhc/dev/octavolabs/vernemq'
strip:_build/default/rel/vernemq/bin/cuttlefish: file format not recognized
strip:_build/default/rel/vernemq/bin/vmq-admin: file format not recognized
strip:_build/default/rel/vernemq/bin/vernemq: file format not recognized
strip:_build/default/rel/vernemq/lib/inets-7.0.6/priv/bin/runcgi.sh: file format not recognized
strip:_build/default/rel/vernemq/lib/env.sh: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/cuttlefish: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/start_erl.src: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/nodetool: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/erl.src: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/start: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/erl: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/start.src: file format not recognized
dpkg-shlibdeps: error: cannot read debian/control: No such file or directory
xargs: dpkg-shlibdeps: exited with status 255; aborting
# required for running dpkg-shlibdeps
mkdir -p debian
touch debian/control
fpm -s dir -t deb -v "1.7.0-34-g4174d018" \
	--force \
	--name vernemq \
	--epoch 1553699570 \
	--license "Apache 2.0" \
	--url "https://vernemq.com" \
	--vendor "Octavo Labs AG" \
	--maintainer "<info@vernemq.com>" \
	--description "VerneMQ is a MQTT message broker" \
	--depends logrotate \
	--depends sudo \
	--depends adduser \
	--depends "" \
	--deb-user vernemq \
	--deb-group vernemq \
	--deb-changelog ./changelog.md \
	--deb-no-default-config-files \
	--deb-systemd files/vernemq.service \
	--after-install files/deb-vernemq.postinst \
	--config-files /etc/vernemq/vernemq.conf \
	_build/default/rel/vernemq/bin/vernemq=/usr/bin/vernemq \
	_build/default/rel/vernemq/bin/vmq-admin=/usr/bin/vmq-admin \
	_build/default/rel/vernemq/data=/var/lib/vernemq/ \
	_build/default/rel/vernemq/etc/=/etc/vernemq/ \
	_build/default/rel/vernemq/bin/=/usr/lib/vernemq/bin/ \
	_build/default/rel/vernemq/lib=/usr/lib/vernemq/ \
	_build/default/rel/vernemq/releases=/usr/lib/vernemq/ \
	_build/default/rel/vernemq/erts-"10.3"=/usr/lib/vernemq/ \
	_build/default/rel/vernemq/share/=/usr/share/vernemq/ \
	_build/default/rel/vernemq/log/=/var/log/vernemq/
Adding action files
epoch in Version is set {:epoch=>"1553699570", :level=>:warn}
Created package {:path=>"vernemq_1.7.0-34-g4174d018_amd64.deb"}
21.3.1 lhc@algol ~/dev/octavolabs/vernemq (use-fpm-instead-of-nodepackage) $ sudo dpkg -i ./vernemq_1.7.0-34-g4174d018_amd64.deb 
[sudo] password for lhc: 
dpkg: error processing archive ./vernemq_1.7.0-34-g4174d018_amd64.deb (--install):
 parsing file '/var/lib/dpkg/tmp.ci/control' near line 8 package 'vernemq':
 'Depends' field, missing package name, or garbage where package name expected
Errors were encountered while processing:
 ./vernemq_1.7.0-34-g4174d018_amd64.deb